zeppelin-web/src/app/notebook/notebook.html

```diff
@@ -115,9 +120,9 @@
         <i class="icon-question"></i>
       </span>
       <span style="position:relative; top:2px; margin-right:4px; cursor:pointer;"
-            ng-click="toggleSetting()"
+            ng-click="!isReadOnly() && toggleSetting()"
             tooltip-placement="bottom" tooltip="Interpreter binding">
-        <i class="fa fa-cog" ng-style="{color: showSetting ? '#3071A9' : 'black' }"></i>
+        <i class="fa fa-cog" ng-style="{color: showSetting ? '#3071A9' : 'black' }" ng-class="{'disabled':isReadOnly()}"></i>
```

Comment:
Setting button in 'disabled' class doesn't look different much.
![image](https://cloud.githubusercontent.com/assets/1540981/11161259/3ec10c1c-8abb-11e5-9274-93f478a3c4e2.png)

I think it's better hide the button.